src/vs/workbench/browser/positronModalDialogs/copyAsCodeModalDialog.tsx

```diff
@@ -0,0 +1,164 @@
+import { OKCancelModalDialog } from '../positronComponents/positronModalDialog/positronOKCancelModalDialog.js';
+import { IPositronDataExplorerInstance } from '../../services/positronDataExplorer/browser/interfaces/positronDataExplorerInstance.js';
+import { PositronDataExplorerCommandId } from '../../contrib/positronDataExplorerEditor/browser/positronDataExplorerActions.js';
+import { DropDownListBox } from '../positronComponents/dropDownListBox/dropDownListBox.js';
+import { DropDownListBoxItem } from '../positronComponents/dropDownListBox/dropDownListBoxItem.js';
+import { DropdownEntry } from './components/dropdownEntry.js';
+
+/**
+ * Shows the new folder from Git modal dialog.
+ * @param commandService The command service.
+ * @param keybindingService The keybinding service.
+ * @param layoutService The layout service.
+ * @param dataExplorerClientInstance The data explorer client instance.
+ * @returns A promise that resolves when the dialog is closed.
+ */
+export const copyAsCodeModalDialog = async (
+	commandService: ICommandService,
+	keybindingService: IKeybindingService,
+	layoutService: IWorkbenchLayoutService,
+	dataExplorerClientInstance: IPositronDataExplorerInstance,
+): Promise<void> => {
+	// Create the renderer.
+	const renderer = new PositronModalReactRenderer({
+		keybindingService,
+		layoutService,
+		container: layoutService.activeContainer
+	});
+
+	// Show the copy as code dialog.
+	renderer.render(
+		<NewCopyAsCodeModalDialog
+			commandService={commandService}
+			dataExplorerClientInstance={dataExplorerClientInstance}
+			keybindingService={keybindingService}
+			layoutService={layoutService}
+			renderer={renderer}
+		/>
+	);
+};
+
+/**
+ * CopyAsCodeDialogProps interface.
+ */
+interface CopyAsCodeDialogProps {
+	commandService: ICommandService;
+	dataExplorerClientInstance: IPositronDataExplorerInstance
+	keybindingService: IKeybindingService;
+	layoutService: IWorkbenchLayoutService;
+	renderer: PositronModalReactRenderer;
+}
+
+
+/**
+ * NewCopyAsCodeModalDialog component.
+ * @param props The component properties.
+ * @returns The rendered component.
+ */
+export const NewCopyAsCodeModalDialog = (props: CopyAsCodeDialogProps) => {
+	// State hooks.
+	const instance = props.dataExplorerClientInstance.dataExplorerClientInstance;
+	const codeSyntaxOptions = instance.cachedBackendState?.supported_features?.code_syntaxes ?? [];
+
+	const [selectedSyntax, setSelectedSyntax] = useState<string>(instance.guessedSyntax.code_syntax ?? 'Select Code Syntax');
+
+	const [codeString, setCodeString] = useState<string | undefined>(undefined);
+
+	useEffect(() => {
+		const getCodeString = async () => {
+			if (!selectedSyntax) {
+				return;
+			}
+			// Execute the command to get the code string based on the selected syntax.
+			const codeString = await props.commandService.executeCommand<string>(PositronDataExplorerCommandId.CopyAsCodeAction, selectedSyntax);
```

Comment:
We don't need the return typing here - that gets inferred by TypeScript.
```suggestion
			const codeString = await props.commandService.executeCommand(PositronDataExplorerCommandId.CopyAsCodeAction, selectedSyntax);
```